src/modules/colorPicker/ColorPickerUI/Helpers/ColorHelper.cs

```diff
@@ -139,6 +139,69 @@ internal static (string hue, double whiteness, double blackness) ConvertToNatura
             return (GetNaturalColorFromHue(color.GetHue()), min, 1 - max);
         }
 
+        /// <summary>
+        /// Convert a given <see cref="Color"/> to a CIE LAB color (LAB)
+        /// </summary>
+        /// <param name="color">The <see cref="Color"/> to convert</param>
+        /// <returns>The lightness [0..100] and two chromaticities [-128..127]</returns>
+        internal static (double lightness, double chromaticityA, double chromaticityB) ConvertToCIELABColor(Color color)
+        {
+            var xyz = ConvertToCIEXYZColor(color);
+            var lab = GetCIELABColorFromCIEXYZ(xyz.x, xyz.y, xyz.z);
+
+            return lab;
+        }
+
+        /// <summary>
+        /// Convert a given <see cref="Color"/> to a CIE XYZ color (XYZ)
+        /// </summary>
+        /// <param name="color">The <see cref="Color"/> to convert</param>
+        /// <returns>The X [0..1], Y [0..1] and Z [0..1]</returns>
+        internal static (double x, double y, double z) ConvertToCIEXYZColor(Color color)
+        {
+            double r = color.R / 255d;
+            double g = color.G / 255d;
+            double b = color.B / 255d;
+
+            // inverse companding, gamma correction must be undone
+            double rLinear = (r > 0.04045) ? Math.Pow((r + 0.055) / 1.055, 2.4) : (r / 12.92);
+            double gLinear = (g > 0.04045) ? Math.Pow((g + 0.055) / 1.055, 2.4) : (g / 12.92);
+            double bLinear = (b > 0.04045) ? Math.Pow((b + 0.055) / 1.055, 2.4) : (b / 12.92);
+
+            return (
+                (rLinear * 0.4124) + (gLinear * 0.3576) + (bLinear * 0.1805),
+                (rLinear * 0.2126) + (gLinear * 0.7152) + (bLinear * 0.0722),
+                (rLinear * 0.0193) + (gLinear * 0.1192) + (bLinear * 0.9505)
+            );
+        }
+
+        /// <summary>
+        /// Convert a CIE XYZ color <see cref="double"/> to a CIE LAB color (LAB)
```

Comment:
Can you add some link into some documentation that specify all these constants? (At least a wiki page), right now it looks like a bunch of random numbers, the same for the previous function

Reply:
Thanks a lot for your feedback, I really appreciate it 😄!

For converting from sRGB to XYZ, I used, among others, this page: https://en.wikipedia.org/wiki/SRGB#The_reverse_transformation_(sRGB_to_CIE_XYZ)

For converting from XYZ to L*a*b*, I used, among others, this page: https://en.wikipedia.org/wiki/CIELAB_color_space#Converting_between_CIELAB_and_CIEXYZ_coordinates.

Should I refer these pages in the code as well (in the summary)?